### PR TITLE
policy: minor OutboundPolicy gRPC server refactors 

### PR DESCRIPTION
**Pull request by hawkw**
This branch makes a few very minor style tweaks to the OutboundPolicy
API server, primarily in the handling of authority strings in the
`lookup_authority` function. There should be no functional changes in
this branch.

### CODE REVIEW

File: policy-controller/grpc/src/outbound.rs [hunk 1]
```diff
@@ -154,7 +151,7 @@ where
             .index
             .watch_outbound_policy(service)
             .await
-            .map_err(|e| tonic::Status::internal(format!("lookup failed: {}", e)))?
+            .map_err(|e| tonic::Status::internal(format!("lookup failed: {e}",)))?
```
Comment (mateiidavid) [inline policy-controller/grpc/src/outbound.rs:154]: super nitpicky but could probably remove the comma.

```suggestion
            .map_err(|e| tonic::Status::internal(format!("lookup failed: {e}")))?
```

**Review by mateiidavid — APPROVED** [review]

**Comment by hawkw** [comment]
Looks like the CI docker build keeps timing out on this branch: https://github.com/linkerd/linkerd2/actions/runs/4439310495/jobs/7791651346?pr=10547#step:3:1090 :/

**Review by adleong — APPROVED** [review]
For my education: when is it appropriate to use `{}", foo` style vs `{foo}"` in format strings?

**Comment by hawkw** [comment]
> For my education: when is it appropriate to use `{}", foo` style vs `{foo}"` in format strings?

If an expression is the name of a local variable, it can be directly inlined into the format string. Using `"{}", <EXPR>` is necessary when the expression is more complex than an identifier in the current scope, like `"{}", foo.bar().baz` or similar.

**Pull request merged**